web/src/api/posts/GetAllPosts.ts

```diff
@@ -0,0 +1,17 @@
+import Post from '@/models/posts/Post';
+import APIResponse from '../APIResponse';
+import ErrorHandler from '../services/ErrorHandler';
+import { fetchGet } from '../utils/fetch';
+
+const PATH = '/post';
+
+const GetAllPosts = async (): Promise<APIResponse<Post>> => {
+    try {
+        const response = await fetchGet<{}>(PATH, {});
```

Comment:
Should this be optional rather than requiring an empty body? I feel like it might be better to leave it as undefined and use coalesce to convert it to an empty object, if needed.

Edit: Oh, it is optional. Maybe the generic is not optional. If not, make it <T = undefined> or whatever the letter is.